UserScrape/crawler.py

```diff
@@ -321,6 +326,69 @@ async def watch_videos(self, videos: list[str]):
+        # on the home page, so we have to get rid of them
+        # if we close these extra sections, YouTube remembers and doesnt show them again
+        # ideally, this loop is only run once per account
+        feed_is_bannerfree = False
+        while not feed_is_bannerfree:
+            # refresh the feed everytime we had to close something until we finally get a completely clean feed
+            self.driver.get("https://www.youtube.com")
+            # set the stop condition to True unless any 'banners' are detected
+            feed_is_bannerfree = True
+            try:
+                # this is the link to the WHO
+                # there are 3 other buttons with the same aria-label, which cannot be clicked, so I simply try to click them all and catch the exception
+                # unfortunately there is no other way to uniquely identify that button
+                extra_content = WebDriverWait(self.driver, 2).until(  # Schließen
+                    EC.presence_of_all_elements_located((By.XPATH, '//*[@aria-label="Close"]')))
+                for button in extra_content:
+                    try:
+                        button.click()
+                        feed_is_bannerfree = False
+                        print("information closed")
+                    except ElementNotInteractableException:
+                        pass
+                    except ElementNotVisibleException:
+                        pass
+            except TimeoutException:
+                # print("no extra covid information")
+                pass
+
+            # these kinds of banners are partly corona specific (like #fitnessathome) or not (#trendingmovies)
+            themed_content = None
+            try:
+                themed_content = WebDriverWait(self.driver,2).until(  # Kein Interesse
+                    EC.presence_of_all_elements_located((By.XPATH, '//*[@aria-label="Not interested"]')))
+            except TimeoutException:
+                # print("No themed content")
+                pass
+
+            if themed_content is not None:
+                feed_is_bannerfree = False
+                for button in themed_content:
+                    button.click()
+                    print("banner closed")
+
+        all_videos = WebDriverWait(self.driver, 10).until(
+            EC.presence_of_all_elements_located((By.XPATH, '//*[@id="video-title-link"]'))
+        )
+
+        feed_videos = []
+        for video in all_videos:
+            # take the link and remove everything except for the id of the video that the link leads to
+            recommendation_id = video.get_attribute('href').replace('https://www.youtube.com/watch?v=', '')
+            feed_videos.append(recommendation_id)
+
+        filename = 'output/feed/' + self.email + '_' + \
+        str(self.init_time).replace(':', '-').replace(' ', '_') + '.txt'
+        # upload the information as a blob
+        self.__save_file(seshPath / filename, str(feed_videos))
```

Comment:
How shall this be stored? Right now I just write a list to a file...

Reply:
Simple option will be just like this, but with json (insteaf o txt) that also includes the video title & channel. Even tho we can look up the video in our dataset, if its missing/deleted it is helpfull to know.

Side note: I have been thinking about confluence for this, but given how tiny our volume of results are I think its simpler & acceptable to wait for a blob file to be written and aggregate them together later.